src/cares.c

```diff
@@ -486,16 +489,30 @@ query_txt_cb(void *arg, int status,int timeouts, unsigned char *answer_buf, int
         goto callback;
     }
 
+    tmp_obj = PyStructSequence_New(&AresQueryTXTResultType);
```

Comment:
you can set these to null and create them inside the loop as in the CFFI variant. Also: error checking.